src/annotator/anchoring/pdf.js

```diff
@@ -347,6 +372,7 @@ export function anchor(root, selectors) {
   const position = selectors.find(s => s.type === 'TextPositionSelector');
   const quote = selectors.find(s => s.type === 'TextQuoteSelector');
 
+  /** @type {Promise<Range>} */
   let result = Promise.reject('unable to anchor');
```

Comment:
I'm not sure I've ever quite seen this pattern: reject and then catch later if things are not problematic...intriguing...

Reply:
As historical context, this code was written before `async`/`await` support existed in our codebase. It could be written in a more straightforward way now using that language feature.